android/demo-app/src/main/java/com/stadiamaps/ferrostar/DemoNavigationViewModel.kt

```diff
@@ -24,7 +25,8 @@ import uniffi.ferrostar.UserLocation
 class DemoNavigationViewModel(
     // This is a simple example, but these would typically be dependency injected
     private val ferrostarCore: FerrostarCore = AppModule.ferrostarCore,
-) : ViewModel(), LocationUpdateListener, NavigationViewModel {
+    annotationPublisher: AnnotationPublisher<*> = valhallaExtendedOSRMAnnotationPublisher()
+) : DefaultNavigationViewModel(ferrostarCore, annotationPublisher), LocationUpdateListener {
```

Comment:
@ianthetechie now that this view model uses the `DefaultNavigationViewModel`. We should be able to refactor it to use the existing uiState from ferrostar and implement its own additional state for things like route loading, etc. Just to highlight how easy it is to customize without interrupting the default navigation behavior.